code/parachain/frame/composable-traits/src/staking/lock.rs

```diff
@@ -16,13 +16,16 @@ use crate::time::{DurationSeconds, Timestamp};
 #[scale_info(skip_type_params(MaxDurationPresets))]
 pub struct LockConfig<MaxDurationPresets: Get<u32>> {
 	/// The possible locking duration.
+	// TODO(benluelo): This should be a curve (but linear), with start and end points, and a
```

Comment:
Why go for a curve when there is no curve(staking rewards are discrete functions) requirement for rewards? Also per second assumes that rewards can always be calculated per second which is not the case as per product requirement. We can have per week rewards if needed.